ml/base/estimator.py

```diff
@@ -0,0 +1,55 @@
+import tensorflow as tf
+
+
+class MLEstimator(object):
+    def __init__(self, config, graph, trainer, evaluator, metrics):
+        self.config = config
+        self.graph = graph
+        self.trainer = trainer
+        self.evaluator = evaluator
+        self.metrics = metrics
+        self.estimator = self._gen_estimator()
+
+        self.output = None
+        self.labels = None
+        self.loss = None
+
+    @classmethod
+    def from_config(cls, config):
+        graph = Graph.from_config(config.graph)
+        trainer = Trainer.from_config(config.trainer)
+        evaluator = Evaluator.from_config(config.evaluator)
+        cls(config, model, trainer, evaluator)
+
+    def _gen_estimator(self):
+        estimator = tf.estimator.Estimator(
+            model_fn=self._gen_model_fn(),
+            input_fn=self._gen_input_fn(),
+        )
+        return estimator
+
+    def _gen_model_fn(self):
+        def model_fn(features, labels, mode, params):
```

Comment:
`params` 可以怎么用? `params` 是不是可以是config的一个默认field?

Reply:
params is basically the place to store all hyperparameters of the model, but since we have our own way of storing them (in config and graph) so we simply don't use it.